File: LeetCode/1629. Slowest Key/solution.py
class Solution:
    def slowestKey(self, releaseTimes: List[int], keysPressed: str) -> str:
        arr = []
        n = len(releaseTimes)
        for i in range(n):
            if i > 0:
                arr.append((releaseTimes[i]-releaseTimes[i-1], keysPressed[i]))
            else:
                arr.append((releaseTimes[i],keysPressed[i]))
        arr.sort()
        # print(arr)
        return arr[-1][-1]

File: LeetCode/1629. Slowest Key/solution.cpp
class Solution {
public:
    char slowestKey(vector<int>& releaseTimes, string keysPressed) {
        vector<int> tik(26,0);
        int res = 0, idx = -1, pre = 0, n = releaseTimes.size();
        for (int i = 0; i < n; i++){
            int cur = releaseTimes[i];
            char ch = keysPressed[i];
            if (cur-pre > tik[ch-'a'])
                tik[ch-'a'] = cur-pre;
            pre = cur;
        }
        for (int i = 25; i >= 0; i--){
            if (tik[i] > res){
                res = tik[i];
                idx = i;
            }
        }
        return 'a'+idx;
    }
};